lightning/src/ln/channelmanager.rs

```diff
@@ -2718,6 +2757,16 @@ impl<ChanSigner: ChannelKeys, M: Deref + Sync + Send, T: Deref + Sync + Send, K:
 		}
 		self.latest_block_height.store(height as usize, Ordering::Release);
 		*self.last_block_hash.try_lock().expect("block_(dis)connected must not be called in parallel") = header_hash;
+		loop {
+			// Just in case we end up in a race, we loop until we either successfully update
+			// last_node_announcement_serial or decide we don't need to.
+			let old_serial = self.last_node_announcement_serial.load(Ordering::Acquire);
+			if old_serial < header.time as usize {
+				if self.last_node_announcement_serial.compare_exchange(old_serial, header.time as usize, Ordering::AcqRel, Ordering::Relaxed).is_ok() {
+					break;
+				}
+			} else { break; }
+		}
```

Comment:
Confused about this loop -- we don't broadcast a NodeAnnouncement in this function, so what's the point of updating `last_node_announcement_serial`?

Reply:
The `compare_exchange` updates it - the goal here is to update the value iff the block timestamp is >= the current latest value. I updated the comment to note that. Rust does have a method for this, but sadly its nightly-only (and it should compile down to something similar, just maybe with more optimal/effecient Orderings.

Reply:
Ah, so the purpose is to have a "fresh" timestamp for the next time we _do_ broadcast a node announcement